skbeam/core/utils.py

```diff
@@ -1271,3 +1271,36 @@ def bin_grid(image, r_array, pixel_sizes, statistic='mean', mask=None,
     bin_centers = bin_edges_to_centers(bin_edge)
 
     return bin_centers, int_stat
+
+
+def time_binning(time_diff, ff=2):
+    """
+    Bin the time differences into all possible bins
+
+    Parameters
+    ----------
+    time_diff : list
+        time difference from previous image for each image as a list
+    ff : int, optional
+        desired float point format
+
+    Returns
+    -------
+    time_bin : list
+        time bin for each delay time according to time bins
+    all_lags : array
+        all possible time lags from the time difference
+
+    """
+    time_bins = []
+    all_lags = []
+    ff_p = '.' + str(ff) + 'f'
+    for c, t in enumerate(time_diff):
+        stamp = 0
+        time_b = []
+        for item in (time_diff[:c+1][::-1]):
+            time_b.append(float(format(stamp, ff_p)))
```

Comment:
@danielballan Thanks for the comments. I don't think `np.diigtize` will work here because I don't have a control of the input array to bin and the array of bins. Here I am converting result to string to get rid of floating point issues. This will be very helpful for one-time correlation functions with variable time stamps. Any better ways than this?